jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/ByteAccumulator.java

```diff
@@ -43,11 +72,7 @@ public void copyChunk(byte[] buf, int offset, int length)
             String err = String.format("Resulting message size [%,d] is too large for configured max of [%,d]", this.length + length, maxSize);
             throw new MessageTooLargeException(err);
         }
-
-        byte[] copy = new byte[length - offset];
-        System.arraycopy(buf, offset, copy, 0, length);
-
-        chunks.add(copy);
+        chunks.add(ByteBuffer.wrap(buf, offset, length));
```

Comment:
This `copyChunk(byte[] buf, int offset, int length)` method doesn't really work with the `ByteBufferPool`. This ByteBuffer will eventually be returned to the pool but we have created it right here. I think this will give an error if it is used.

Reply:
It works with `bufferPool  == null `. if `bufferPool  == null ` when invoke `newByteBuffer` method. that will return a new `ByteBuffer` without `bufferPool` management. and then invoke `copyChunk(byte[] buf, int offset, int length)` that will works fine.